agent/mapcall/src/main/java/randoop/instrument/CallReplacementTransformer.java

```diff
@@ -169,310 +95,328 @@ public CallReplacementTransformer() {}
+      return null;
+    }
+    MethodDef call = getReplacementMethod(orig);
+    if (call == null) {
+      debug_transform.log(
+          "%s.%s: No replacement for %s%n", mg.getClassName(), mg.getName(), orig.toString());
+      debug_transform.log(
+          "getReplacementInstruction: EXIT %s.%s%n", mg.getClassName(), mg.getName());
+      return null;
+    }
 
+    debug_transform.log(
+        "%s.%s: Replacing method %s with %s%n",
+        mg.getClassName(), mg.getName(), orig.toString(), call);
+
+    InvokeInstruction instruction;
+
+    switch (inst.getOpcode()) {
+      case Const.INVOKEDYNAMIC:
+      case Const.INVOKEINTERFACE:
+      case Const.INVOKESPECIAL:
+      case Const.INVOKEVIRTUAL:
+        /*
+         * These are invocations where an implicit argument occurs in the call.
+         */
+        Type instanceType = invocation.getReferenceType(pgen);
+        Type[] arguments = BCELUtil.insert_type(instanceType, invocation.getArgumentTypes(pgen));
+        instruction =
+            ifact.createInvoke(
+                call.getClassname(),
+                call.getName(),
+                invocation.getReturnType(pgen),
+                arguments,
+                Const.INVOKESTATIC);
+        break;
+      case Const.INVOKESTATIC:
+        instruction =
+            ifact.createInvoke(
+                call.getClassname(),
+                call.getName(),
+                invocation.getReturnType(pgen),
+                invocation.getArgumentTypes(pgen),
+                Const.INVOKESTATIC);
+        break;
       default:
-        return (null);
+        debug_transform.log(
+            "getReplacementInstruction: EXIT wrong instruction type in %s.%s%n",
+            mg.getClassName(), mg.getName());
+        return null;
     }
+    debug_transform.log("getReplacementInstruction: EXIT %s.%s%n", mg.getClassName(), mg.getName());
+    return instruction;
   }
 
   /**
-   * Replace instruction ih in list il with the instructions in new_il. If new_il is null, do
-   * nothing.
+   * Returns the replacement method for the given method if one is determined by a method, class or
+   * package replacement, or null otherwise.
    *
-   * @param il the instruction list
-   * @param ih the instruction
-   * @param new_il the new instructions to substitute
+   * @param methodDef the method to replace, must not be null
+   * @return the replacement method, null if there is none
    */
-  protected static void replace_instructions(
-      InstructionList il, InstructionHandle ih, InstructionList new_il) {
+  private MethodDef getReplacementMethod(MethodDef methodDef) {
+    return replacementMap.get(methodDef);
+  }
 
-    if ((new_il == null) || new_il.isEmpty()) {
-      return;
+  /**
+   * Replace the instruction in the original instruction list with the instructions in the new
+   * instruction list. If this instruction was the target of any jumps or line numbers, replace them
```

Comment:
There isn't a list.  There is a single instruction.  So the documentation needs to be updated, and the code too.